src/Slick/Pandoc.hs

```diff
@@ -50,76 +40,99 @@ unPandocM p = do
+--   which implements 'FromJSON'. Failure to deserialize will fail the Shake
+--   build.
+makePandocReader' :: (FromJSON a) => PandocReader textType
+                  -> textType
+                  -> Action (Pandoc, a)
 makePandocReader' readerFunc text = do
   (pdoc, meta)  <- makePandocReader readerFunc text
   convertedMeta <- convert meta
   return (pdoc, convertedMeta)
 
+--------------------------------------------------------------------------------
+
 -- | Load in a source document using the given 'PandocReader', then render the 'Pandoc'
--- into text using the given 'PandocWriter'.
--- Returns a 'Value' wherein the rendered text is set to the "content" key and 
--- any metadata is set to its respective key in the 'Value'
-loadUsing :: PandocReader textType -> PandocWriter -> textType -> Action Value
+--   into text using the given 'PandocWriter'.
+--   Returns a 'Value' wherein the rendered text is set to the "content" key and
+--   any metadata is set to its respective key in the 'Value'
+loadUsing :: PandocReader textType
+          -> PandocWriter
+          -> textType
+          -> Action Value
 loadUsing reader writer text = do
   (pdoc, meta) <- makePandocReader reader text
   outText      <- unPandocM $ writer pdoc
   let withContent = meta & _Object . at "content" ?~ String outText
   return withContent
 
 -- | Like 'loadUsing' but allows also deserializes the 'Value' into any object
--- which implements 'FromJSON'.  Failure to deserialize will fail the Shake
--- build.
-loadUsing'
-  :: (FromJSON a)
-  => PandocReader textType
-  -> PandocWriter
-  -> textType
-  -> Action a
-loadUsing' reader writer text = loadUsing reader writer text >>= convert
-
--- | Attempt to convert between two JSON serializable objects (or 'Value's).
--- Failure to deserialize fails the Shake build.
-convert :: (FromJSON a, ToJSON a, FromJSON b) => a -> Action b
-convert a = case fromJSON (toJSON a) of
-  Success r   -> pure r
-  Error   err -> fail $ "json conversion error:" ++ err
-
--- | Flatten a Pandoc 'Meta' into a well-structured JSON object, rendering Pandoc
--- text objects into plain strings along the way.
-flattenMeta :: Meta -> Value
-flattenMeta (Meta meta) = toJSON $ fmap go meta
- where
-  go :: MetaValue -> Value
-  go (MetaMap     m) = toJSON $ fmap go m
-  go (MetaList    m) = toJSONList $ fmap go m
-  go (MetaBool    m) = toJSON m
-  go (MetaString  m) = toJSON m
-  go (MetaInlines m) = toJSON $ stringify m
-  go (MetaBlocks  m) = toJSON $ stringify m
+--   which implements 'FromJSON'.  Failure to deserialize will fail the Shake
+--   build.
+loadUsing' :: (FromJSON a)
+           => PandocReader textType
+           -> PandocWriter
+           -> textType
+           -> Action a
+loadUsing' reader writer text =
+  loadUsing reader writer text >>= convert
+
+-- | Generalized Entity Loader, wrapper in
+--   own function to respolve EntityFilePath properly
+loadEntity :: FromJSON b
```

Comment:
I'm missing the significance of this; Why the `EntityFilePath` wrapper? I'm assuming it's for use with a cache, but if that's the case we should likely also export a helper which provides such a cache.

Reply:
Maybe we can also rename it as "Document", and then refer to it as the document and its metadata?

Reply:
@ChrisPenner `shake` caches create conversion object based on filepath and if used singled version it can't distinguish between entities. That required specific wrapper `EntityFilePath`on top of it to distinguish entities in a build process. Useful when having multiple objects like`post`, `draft-post`, `presentations`, etc. and separate build rules for them.

Reply:
> if that's the case we should likely also export a helper which provides such a cache.

Currently this is a wrapper which is exposed without being used anywhere, unless there's a clear generalizable and easy to use case for it I think it should be moved from library-land to user-land.

Reply:
@ChrisPenner Chris, I understand concerns about complicating project, so I can take down loaders functionality. As we're trying to provide utilities for companies where many more site entities exists, more additional things required that's why it may seem  opinionated. I was wondering if it's possible to feature our package instead, like link or mention? For example if we make `slick-extra` or `slicker` (not sure which name works better) where this new functions live and add a section in this repositorie's readme for advanced users. As our utilities consist of universal loaders, sitemap, rss, schema generations, live view, and couple more goodies that I believe must be for production website and development workflow.

Reply:
Absolutely! I'm more than happy to add a section to the readme for plugins/extensions and other packages 👍; or feel free to add that yourself if you like